packages/mdc-textfield/helper-text/README.md

```diff
@@ -0,0 +1,133 @@
+indicate to assistive devices that the display of the helper text is dependent on the interaction with
+the input element.
+
+When using our vanilla JS component, if it sees that the input element has an `aria-controls`
+attribute, it will look for an element with the id specified and treat it as the text field's help
+text element, taking care of adding/removing `aria-hidden` and other a11y attributes. This can also
+be done programmatically, which is described below.
+
+### Validation
+
+MDC TextField provides validity styling by using the `:invalid` and `:required` attributes provided
+by HTML5's form validation API.
+
+```html
+<div class="mdc-text-field">
+  <input type="password" id="pw" class="mdc-text-field__input" required minlength=8>
+  <label for="pw" class="mdc-text-field__label">Password</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+```
+
+By default an input's validity is checked via `checkValidity()` on blur, and the styles are updated
+accordingly. Set the MDCTextField.valid variable to set the input's validity explicitly. MDC TextField
+automatically appends an asterisk to the label text if the required attribute is set.
+
+Helper text can be used to provide additional validation messages. Use
+`mdc-text-field-helper-text--validation-msg` to provide styles for using the helper text as a validation
+message. This can be easily combined with `mdc-text-field-helper-text--persistent` to provide a robust
+UX for client-side form field validation.
+
+```html
+<div class="mdc-text-field">
+  <input required minlength=8 type="password" class="mdc-text-field__input" id="pw"
+         aria-controls="pw-validation-msg">
+  <label for="pw" class="mdc-text-field__label">Choose password</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+<p class="mdc-text-field-helper-text
+          mdc-text-field-helper-text--persistent
+          mdc-text-field-helper-text--validation-msg"
+   id="pw-validation-msg">
+  Must be at least 8 characters long
+</p>
+```
+
+#### MDCTextFieldHelperText API
+
+##### MDCTextFieldHelperText.foundation
+
+MDCTextFieldHelperTextFoundation. This allows the parent MDCTextField component to access the public methods on the MDCTextFieldHelperTextFoundation class.
+
+### Using the foundation class
+
+Method Signature | Description
+--- | ---
+addClass(className: string) => void | Adds a class to the helper text element
+removeClass(className: string) => void | Removes a class from the helper text element
+hasClass(className: string) => boolean | Returns true if classname exists for the helper text element
+setAttr(attr: string, value: string) => void | Sets an attribute with a given value on the helper text element
+removeAttr(attr: string) => void | Removes an attribute on the helper text element |
+
+#### The full foundation API
+
+##### MDCTextFieldHelperTextFoundation.show()
+
+Makes the helper text visible to screen readers.
```

Comment:
hmm its really not obvious that show only makes it visible to screen readers, and that making helper text visible to users in general is handle in some other code. Whats a better method name to make that more clear?

Reply:
how about showToScreenReader? some other options: makeVisible, makeVisibleToScreenReader, announce, announceToScreenReader